src/utils/put.ts

```diff
@@ -0,0 +1,55 @@
+import FormData from 'form-data';
+import fs from 'fs-extra';
+import path from 'path';
+import url from 'url';
+
+export default function put(
+  urlPath: string,
+  files: string | string[],
+  headers: Record<string, string> | Callback<any, Error | string>,
+  callback: Callback<any, Error | string>
+) {
+  if (typeof headers === 'function') {
+    callback = headers;
+    headers = {};
+  }
+
+  const form = new FormData();
+  const options = { ...url.parse(urlPath), method: 'PUT', headers: {} };
+  let body = '';
+  let callbackDone = false;
+
+  if (!Array.isArray(files)) {
+    files = [files];
+  }
+
+  files.forEach(file => {
+    const fileName = path.basename(file);
+    form.append(fileName, fs.createReadStream(file));
+  });
+
+  options.headers = { ...headers, ...form.getHeaders() };
+
+  form.submit(options as any, (err, res) => {
```

Comment:
why do we need this any and on line 36?

Reply:
clash betwen libraries, url.parse says protocol can be string | null and form.submit accepts string | undefined
its silly but im leaving anys that require refactors and I will tackle those on a second set of prs

Reply:
ok